proxy/src/outbound.rs

```diff
@@ -80,11 +87,15 @@ where
     /// changed.
     fn bind_service(
         &mut self,
-        authority: &FullyQualifiedAuthority,
+        key: &Self::Key,
     ) -> Result<Self::Service, Self::RouteError> {
-        debug!("building outbound client to {:?}", authority);
+        let &(ref authority, protocol) = key;
```

Comment:
Once again, take it or leave it, but consider destructuring in the method params?